test.js

```diff
@@ -670,6 +670,108 @@ function buildTests (test, sync) {
       }, 0)
     })
   })
+
+  test('mode', (t) => {
+    t.plan(6)
+
+    const dest = file()
+    const mode = 0o444
+    const stream = new SonicBoom({ dest, sync, mode })
+
+    stream.on('ready', () => {
+      t.pass('ready emitted')
+    })
+
+    t.ok(stream.write('hello world\n'))
+    t.ok(stream.write('something else\n'))
+
+    stream.end()
+
+    stream.on('finish', () => {
+      fs.readFile(dest, 'utf8', (err, data) => {
+        t.error(err)
+        t.equal(data, 'hello world\nsomething else\n')
+        t.equal(fs.statSync(dest).mode & 0o777, stream.mode)
+      })
+    })
+  })
+
+  test('mode default', (t) => {
+    t.plan(6)
+
+    const dest = file()
+    const defaultMode = 0o644
+    const stream = new SonicBoom({ dest, sync })
+
+    stream.on('ready', () => {
+      t.pass('ready emitted')
+    })
+
+    t.ok(stream.write('hello world\n'))
+    t.ok(stream.write('something else\n'))
+
+    stream.end()
+
+    stream.on('finish', () => {
+      fs.readFile(dest, 'utf8', (err, data) => {
+        t.error(err)
+        t.equal(data, 'hello world\nsomething else\n')
+        t.equal(fs.statSync(dest).mode & 0o777, defaultMode)
+      })
+    })
+  })
+
+  test('mode 0o666 resulting to default 0o644', (t) => {
+    t.plan(7)
+
+    const dest = file()
+    const defaultMode = 0o644
+    const mode = 0o666
+    const stream = new SonicBoom({ dest, sync, mode })
+
+    stream.on('ready', () => {
+      t.pass('ready emitted')
+    })
+
+    t.ok(stream.write('hello world\n'))
+    t.ok(stream.write('something else\n'))
+
+    stream.end()
+
+    stream.on('finish', () => {
+      fs.readFile(dest, 'utf8', (err, data) => {
+        t.error(err)
+        t.equal(data, 'hello world\nsomething else\n')
+        t.not(fs.statSync(dest).mode & 0o777, stream.mode)
+        t.equal(fs.statSync(dest).mode & 0o777, defaultMode)
```

Comment:
This test demonstrates how passing a mode doesn't always change to the expected permissions.

Testing `fs.open` in my local machine I was getting identical results, so this seems to be the way `fs.open` works at the moment.

Reply:
Thank you this helped a lot, I was quite confused! 😅